queries/rust/highlights.scm

```diff
@@ -36,13 +40,30 @@
  (#match? @constant "^[A-Z]"))
 
 ; Assume that uppercase names in paths are types
+(scoped_identifier
+  path: (identifier) @namespace)
+(scoped_identifier
+ (scoped_identifier
+  name: (identifier) @namespace))
+(scoped_type_identifier
+  path: (identifier) @namespace)
+(scoped_type_identifier
+ (scoped_identifier
+  name: (identifier) @namespace))
```

Comment:
It's possible that it was a draft to wait for #453 .
As far as I understood you can do
```lisp
[(scoped_identifier
  path: (identifier) @namespace)
(scoped_type_identifier
  path: (identifier) @namespace)]
```
but not 

```lisp
([scoped_identifier scoped_type_identifier]
  path: (identifier) @namespace)
```
after the emacs tree-sitter creator complained. This is partially addressed by matching on supernodes.